app/scripts/services/fighter.js

```diff
@@ -0,0 +1,179 @@
+"use strict";
+
+function DiceRoller(options) {
+  // Default options and mechanics for the 1d20 strategy.
+  //
+  // May be overriden with an options hash provided as an argument to the dice
+  // roller.
+  var opts = _.extend({
+    difficulty: 10
+  }, options);
+  var _strategy = function() {
+    return _.random(1, 20);
+  };
+  var _criticalMax = 20;
+  var _criticalMin = 1;
+
+  return {
+    /**
+     * Set a specific rolling strategy.
+     *
+     * The default strategy is 1d20.
+     */
+    set strategy(strategyFn) {
+      _strategy = strategyFn;
+    },
```

Comment:
Here, I used [getters and setters](http://javascriptplayground.com/blog/2013/12/es5-getters-setters/), a nice feature from JavaScript not commonly used.